flake.nix

```diff
@@ -1,126 +1,347 @@
+      url = "github:loco-3d/linear-feedback-controller";
+      flake = false;
+    };
+    src-linear-feedback-controller-msgs = {
+      url = "github:loco-3d/linear-feedback-controller-msgs";
+      flake = false;
+    };
+    src-toolbox-parallel-robots = {
+      url = "github:gepetto/toolbox-parallel-robots";
+      flake = false;
+    };
+
+    # keep-sorted end
   };
   outputs =
-    { nixpkgs, self, ... }@inputs:
-    inputs.nix-ros-overlay.inputs.flake-utils.lib.eachDefaultSystem (
-      system:
-      let
-        pkgs = import ./patched-nixpkgs.nix {
-          inherit nixpkgs system;
-          overlays = [
-            inputs.nix-ros-overlay.overlays.default
-            (_super: prev: {
-              gepetto-viewer = prev.gepetto-viewer.overrideAttrs {
-                inherit (inputs.gepetto-viewer.packages.${system}.gepetto-viewer) src;
-              };
-            })
-          ];
-          patches = [
-            inputs.patch-brax
-            inputs.patch-hpp
+    inputs:
+    let
+      pkgsForPatching = inputs.nixpkgs.legacyPackages.x86_64-linux;
+      patches = [
+        # sort this by patch application order
+        # not alphabetically
+        inputs.patch-crocoddyl
+        inputs.patch-aligator
+        inputs.patch-brax
+        inputs.patch-hpp
+        inputs.patch-mim-solvers
+        inputs.patch-osg
+        inputs.patch-piqp
+      ];
+      patchedNixpkgs = (
+        pkgsForPatching.applyPatches {
+          inherit patches;
+          name = "patched nixpkgs";
+          src = inputs.nixpkgs;
+        }
+      );
+      overlay = (
+        final: prev:
+        {
+          inherit (inputs)
+            # keep-sorted start
+            src-agimus-msgs
+            src-colmpc
+            src-linear-feedback-controller
+            src-linear-feedback-controller-msgs
+            # keep-sorted end
+            ;
+          # keep-sorted start block=yes
+          franka-description = prev.rosPackages.humble.franka-description.overrideAttrs {
+            src = inputs.src-franka-description;
+            # somehow depends on python3.12-ros-humble-joint-state-publisher-gui
+            # which depends on ros-humble-python-qt-binding
+            # which depends on pyside2
+            # which is broken on darwin
+            meta.broken = final.stdenv.hostPlatform.isDarwin;
+          };
+          gepetto-viewer = prev.gepetto-viewer.overrideAttrs {
+            src = inputs.src-gepetto-viewer;
+          };
+          # keep-sorted end
+          pythonPackagesExtensions = prev.pythonPackagesExtensions ++ [
+            (
+              python-final: python-prev:
+              {
+                inherit (inputs)
+                  # keep-sorted start
+                  src-example-parallel-robots
+                  src-toolbox-parallel-robots
+                  # keep-sorted end
+                  ;
+                agimus-msgs = python-final.toPythonModule final.agimus-msgs;
```

Comment:
```nix
                agimus-msgs = python-final.toPythonModule final.agimus-msgs;
```
Where does python-final come from? the nix language? Like `<pacakge-name>-final`?

Reply:
overlays are functions that take 2 arguments: the final and the previous iteration of the argset.
here, python-final is the first arg of the python overlay, defined l. 135, and final the first arg of the pkgs overlay defined l.110

Reply:
ref. https://wiki.nixos.org/wiki/Overlays#Data_flow_of_overlays